drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -130,43 +132,38 @@ static const uint8_t dbm_to_rx_sens[] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
                                           0x0d, 0x0e, 0x0e, 0x0e, 0x0f };
 #endif
 
-uint16_t at86rf2xx_get_addr_short(const at86rf2xx_t *dev)
+void at86rf2xx_get_addr_short(const at86rf2xx_t *dev, network_uint16_t *addr)
 {
-    return (dev->netdev.short_addr[0] << 8) | dev->netdev.short_addr[1];
+    memcpy(addr, dev->netdev.short_addr, IEEE802154_SHORT_ADDRESS_LEN);
```

Comment:
```suggestion
    memcpy(addr->u8, dev->netdev.short_addr, sizeof(addr->u8));
```

or alternatively even just

```suggestion
    addr->u8[0] = dev->netdev.short_addr[0];
    addr->u8[1] = dev->netdev.short_addr[1];
```

for more type safety?

Reply:
All those three are guaranteed to have the same result: A pointer to a union or struct will always be the same as the pointer to its first member. And modern compilers rarely generate the call to memcpy if the size is known at compile time, but instead generates the most efficient instruction sequence to do the copying.

Reply:
I updated to use `sizeof(*addr)` here. But I didn't change the pointer passed to `memcpy()`, as `addr->u8` and `addr` are guaranteed to have the same value and `memcpy()` uses `void *` anyway.